datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -162,12 +173,30 @@ func (f *ChainDataFetcher) resLoop() {
 				logger.Error("db insertion is failed", "blockNumber", res.blockNumber, "reqType", res.reqType, "err", res.err)
 				// TODO-ChainDataFetcher add retry logic when data insertion is failed
 			} else {
-				f.updateCheckpoint(res.blockNumber.Uint64())
+				if err := f.updateCheckpoint(res.blockNumber.Int64()); err != nil {
+					logger.Error("Failed to update checkpoint", "checkpoint", res.blockNumber.Int64())
+				}
 			}
 		}
 	}
 }
 
-func (f *ChainDataFetcher) updateCheckpoint(num uint64) {
-	// TODO-ChainDataFetcher add logic to update new checkpoint
+func (f *ChainDataFetcher) updateCheckpoint(num int64) error {
+	f.checkpointMap[num] = struct{}{}
+
+	oldCheckpoint := f.checkpoint
```

Comment:
How about using boolean `updated` flag instead of numeric value?
We will write checkpoint only If `updated` is set `true`.

Reply:
Applied. 05a46805ff1e4bfd28b6811e1a075dfce710d85a